src/systems/remove-object3D-system.js

```diff
@@ -116,6 +124,7 @@ export function removeObject3DSystem(world) {
   cleanupSkyboxes(world);
 
   // Finally remove all the entities we just removed from the eid2obj map
-  entities.forEach(removeFromMap);
-  exitedObject3DQuery(world).forEach(removeFromMap);
+  entities.forEach(removeObjFromMap);
+  exitedObject3DQuery(world).forEach(removeObjFromMap);
+  exitedMaterialQuery(world).forEach(removeFromMatMap);
```

Comment:
(Checking my understanding...) 
It is OK not to call `o.material ? o.material.eid ? removeEntity(world, o.material.eid)` in the `traverseSome` above, because the various `cleanup` functions are going to call `disposeMaterial` as needed. We don't risk missing any material `eid`s. Is that correct?

Reply:
That's the idea yeah. I don't think it would be appropriate to remove a material entity just because an Mesh pointing at that material was removed. We specifically dispose of materials when we are done with them through various different means. All of them should always call `disposeMaterial`, which will call `removeEntity`, and then it will be removed from the map here.